solidity/random-beacon/tasks/staking.ts

```diff
@@ -0,0 +1,110 @@
+import { task, types } from "hardhat/config"
+
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+
+task(
+  "staking",
+  "Stakes T tokens, increases authorization and registers the operator "
+)
+  .addOptionalParam("owner", "Staking owner address", undefined, types.string)
+  .addOptionalParam("provider", "Staking provider", undefined, types.string)
+  .addOptionalParam("operator", "Staking operator", undefined, types.string)
+  .addOptionalParam("amount", "Staking amount", 1_000_000, types.int)
+  .addOptionalParam(
+    "authorization",
+    "Authorization amount",
+    undefined,
+    types.int
+  )
+  .setAction(async (args, hre) => {
+    await setup(hre, args)
+  })
+
+async function setup(
+  hre: HardhatRuntimeEnvironment,
+  args: {
+    owner: string
+    provider: string
+    operator: string
+    amount: BigInteger
+    authorization: BigInteger
+  }
+) {
+  const { ethers, helpers, getNamedAccounts } = hre
+  const { owner, provider, operator, amount, authorization } = args
+
+  const { deployer } = await getNamedAccounts()
+  const { to1e18 } = helpers.number
+  const t = await helpers.contracts.getContract("T")
+  const staking = await helpers.contracts.getContract("TokenStaking")
+  const randomBeacon = await helpers.contracts.getContract("RandomBeacon")
+  const accounts = await new ethers.providers.JsonRpcProvider().listAccounts()
+  const stakeOwner = owner || accounts[1]
+  const stakingProvider = provider || accounts[2]
+  const stakingOperator = operator || accounts[3]
+  const beneficiary = stakeOwner
+  const authorizer = stakeOwner
```

Comment:
IMHO we shouldn't pull the accounts from the provider and use those 1, 2, 3 for staking with this script. We should require the addresses of the accounts to be provided as arguments when running this task.

We will have to register multiple nodes, not just one.

Reply:
I tend to agree with @nkuba. This is supposed to be an abstract task that does not make any assumptions about the accounts available for the provider. All parameters should be supplied from the outside and then, we can reuse this task for testnet deployments as well. Right now we assume that:
- The first account available for the provider is the deployer. This will probably not be true for testnet deployments. The deployer is the account we use in our CI, and this account may not be available outside of CI, or even if it is available outside of CI, then not necessarily in the provisioning environment for testnet, or it may not be the first account there.
- There are at least 4 accounts available for the provider. In the simplest case, we need just one account.

I would always expect the owner to be provided from the outside and if we are going to make any assumptions about defaults, the only assumption I'd make is that if not provided, beneficiary == authorizer == owner. This is the default option suggested in Threshold Dashboard and I see no reason for not using it for a local development environment.

A separate complication I asked about in another comment is that I *think* installation script in the current form allows setting only one client instance. We need more and this was supported by the previous version.

To recap, what I would suggest:
- fetch all accounts available for the provider on `install.sh` level (separate task returning an array?),
- iterate over them, call "mint", "stake", "authorize", "register-operator" for all of them,
- to make it simple for local development, let's assume that stake owner == staking provider == beneficiary == authorizer == operator,
- accept all roles as parameters to individual tasks: this will allow us setting more complicated scenarios on testnet.

Opinions?

Reply:
ok, I can adjust the scripts, but to be clear, providing addresses as parameters is still available in this script and we can do so by: `npx hardhat staking --owner 0x123 --provider 0x234 --operator 0x345 --amount 424242` and it will overwrite the default ones (I assumed that would simplify development). In the `install.sh` we just need to pass these parameters to the `staking.ts`

Reply:
After talking with @nkuba we decided on the following approach:
- `install.sh` can be run once which handles fetching all the libs, unlocks, builds and deploying contracts 
- `initialize.sh` should be run for each client. If you need 3 clients, then this script should be run 3 times. `--stake-owner` param is a mandatory param. Other parameters are optional. 

[acd4aeb](https://github.com/keep-network/keep-core/pull/3026/commits/acd4aeb4f3f0c44ed64babd2070f8f20b88c8632)